File: src/AirWaterStore/Services/AirWaterStore/AirWaterStore.API/Models/Message.cs
namespace AirWaterStore.API.Models;

public class Message
{
    public int MessageId { get; set; }

    public int ChatRoomId { get; set; }

    public int UserId { get; set; }

    public string Content { get; set; } = null!;

    public DateTime? SentAt { get; set; } = DateTime.UtcNow;

    public virtual ChatRoom ChatRoom { get; set; } = null!;

    public virtual User User { get; set; } = null!;
}

File: src/AirWaterStore/Services/Chat/Chat.API/app/models/message.py
from pydantic import Field
from datetime import datetime, timezone
from beanie import Document, Indexed


class Message(Document):
    chat_room_id: str
    user_id: int
    content: str
    sent_at: datetime = Field(default_factory=lambda: datetime.now(timezone.utc))

    class Settings:
        name = "messages"
        indexes = [
            ["chat_room_id", "sent_at"],
            "sent_at",
        ]
